### PR TITLE
api: Fixes to allow automated testing and a small bug fix

### PR DESCRIPTION
**Pull request by mattyw**

### CODE REVIEW

File: api/api_test.go [hunk 1]
```diff
@@ -252,6 +256,7 @@ func (s *apiSuite) TestNotFoundError(c *gc.C) {
 func (s *apiSuite) TestSignedAgreementsEnvTermsURL(c *gc.C) {
 	cleanup := testing.PatchEnvironment("JUJU_TERMS", "http://example.com")
```
Comment (cmars) [inline api/api_test.go]: Consider moving these *EnvTermsURL tests to a separate test suite? They share common setup.

**Comment by cmars** [comment]
LGTM

**Comment by alesstimec** [comment]
LGTM

**Comment by mattyw** [comment]
$$merge$$

**Pull request merged**